core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java

```diff
@@ -266,8 +267,23 @@ private synchronized void loadStoredKeysAndFileNames() throws ObjectStoreExcepti
             for (int i = 0; i < files.length; i++)
             {
                 File file = files[i];
-                StoreValue<T> storeValue = deserialize(file);
-                realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                try
+                {
+                    StoreValue<T> storeValue = deserialize(file);
+                    realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                }
+                catch (ObjectStoreException e)
+                {
+                    if (logger.isWarnEnabled())
+                    {
+                        logger.warn(String.format(
+                                "Could not deserialize the ObjectStore file: %s. The file will be skipped and moved " +
+                                "to the Garbage folder", file.getName()));
+                    }
+                    String garbageFileName = file.getParentFile().getParentFile().getParent() + File.separator +
```

Comment:
isn't it best if you just calculate the delta between the app.home and the actual file. This logic is very fragile when it comes to changes in the deployment structure